dashboard/src/components/AppView/ResourceTable/ResourceTable.tsx

```diff
@@ -83,17 +83,25 @@ function ResourceTable({ id, title, resourceRefs }: IResourceTableProps) {
     flattenResources(resourceRefs, state.kube.items),
   );
 
-  const columns = useMemo(() => getColumns(resourceRefs[0]), [resourceRefs]);
+  const columns = useMemo(
+    () => (resourceRefs.length ? getColumns(resourceRefs[0]) : OtherResourceColumns),
+    [resourceRefs],
+  );
   const data = useMemo(
     () =>
-      resources.map((resource, index) =>
-        getData(
-          resourceRefs[index].name,
-          columns.map(c => c.accessor),
-          columns.map(c => c.getter),
-          resource,
-        ),
-      ),
+      resources.map((resource, index) => {
+        // When the resourceRef is a list, the list of references will be just one
+        const ref = resourceRefs[index] || resourceRefs[0];
```

Comment:
This looks a little ambiguous to me, in that if there were 3 resource refs and 5 resources it would get the refs for the first 3 iterations and then the 0th for the fourth and fifth (not that that's possible for the data you expect). If you expect that the length of resourceRefs will be 1 specifically in this case, why not make that explicit:

```
const ref = (resourceRefs.length === 1) ? resourceRefs[0] : resourceRefs[index];  
```
so that if our expectations aren't met, we'll still get an error as we should, rather than masking it with odd behaviour?

Reply:
Got it, yes, it's also more readable that way.